R/ancova.R

```diff
@@ -1,18 +1,49 @@
 
 #' Analysis of Covariance
 #'
-#' Performs an analysis of covariance between two groups returning the estimated "treatment effect"
-#' as well as the least squared means of each group.
+#' Performs an analysis of covariance between two groups returning the estimated
+#' "treatment effect" and least squared means estimates of each group.
 #'
-#' @param data A dataframe containing the variables to be used in the model
-#' @param vars A named list containing the names of the variables for `group`, `visit`, `outcome` and any `covariates`.
+#' @param data A data.frame containing the data to be used in the model
+#' @param vars A vars object as generated by [set_vars()]. Only the `group`,
+#' `visit`, `outcome` and `covariates` elements are required. See details.
+#' @param visits an optional character vector specifying which visits to
+#' fit the ancova model at. If `NULL` all available
+#' visits (as determined by `unique(data[[vars$visit]])`) will be used.
 #' See details.
-#' @param visits an optional character vector specifying which visits to perform ancova at. If `NULL` all all available
-#' visits (as determined by `unique(data[[vars$visit]])`) will be looped over.
 #'
 #' @details
-#' TODO
+#' The function works as follows
 #'
+#' 1. Select the first value from `visits`
+#' 2. Subset the data to only the observations that succoured on this visit
+#' 3. Fit a linear model as  `vars$outcome ~ vars$group + vars$covariates`
+#' 4. Extract the "treatment effect" & least square means for each treatment group
+#' 5. Repeat points 2-3 for all other values in `visits`
+#'
+#' If no value for `visits` are provided then it will be set to
+#' `unique(data[[vars$visit]])`
+#'
+#' In order to meet the formatting standards set by [analyse()] the results
+#' will be collapsed into a single list prefixed by the visit name i.e.
+#' ```
+#' list(
+#'     visit_1_trt = list(est = ...),
+#'     visit_1_lsm_ref = list(est = ...),
+#'     visit_1_lsm_alt = list(est = ...),
+#'     visit_2_trt = list(est = ...),
+#'     visit_2_lsm_ref = list(est = ...),
+#'     visit_2_lsm_alt = list(est = ...),
+#'     ...
+#' )
+#' ```
+#'
+#' The value of covariates provided to [set_vars()] can contain interaction terms
+#' if you want to include interaction terms in your model
+#' i.e. `set_vars(covariates = c("sex*age"))`
```

Comment:
"The value of covariates provided to [set_vars()] can contain interaction terms, e.g. `set_vars(covariates = c("sex*age"))`"

Reply:
done